macaroon.js

```diff
@@ -220,30 +252,48 @@ const base64url = function(buf) {
   Converts a Uint8Array to a bitArray for use by nacl.
   @param {Uint8Array} arr The array to convert.
 */
-function bytesToBitArray(arr) {
-  return sjcl.codec.base64.toBits(nacl.util.encodeBase64(arr));
-}
+const bytesToBits = function(arr) {
+  // See https://github.com/bitwiseshiftleft/sjcl/issues/344 for why
+  // we cannot just use sjcl.codec.bytes.toBits.
+  return sjcl.codec.base64.toBits(naclutil.encodeBase64(arr));
+};
 
 /**
   Converts a bitArray to a Uint8Array.
   @param {bitArray} arr The array to convert.
 */
-function bitArrayToUint8Array(arr) {
-  return nacl.util.decodeBase64(sjcl.codec.base64.fromBits(arr));
-}
+const bitsToBytes = function(arr) {
+  // See https://github.com/bitwiseshiftleft/sjcl/issues/344 for why
+  // we cannot just use sjcl.codec.bytes.toBits.
+  return naclutil.decodeBase64(sjcl.codec.base64.fromBits(arr));
+};
 
 /**
   Converts a hex to Uint8Array
   @param {String} hex The hex value to convert.
   @return {Uint8Array} The resulting array.
 */
-function hexToBytes(hex) {
+const hexToBytes = function(hex) {
   const arr = new Uint8Array(Math.ceil(hex.length / 2));
   for (let i = 0; i < arr.length; i++) {
```

Comment:
could use Array.map here instead?

Reply:
I feel a bit reluctant to use map on arrays that have been created with a capacity - it doesn't work on normal arrays, and it seems a bit dubious by association. I think I'll leave it.

Reply:
:+1: